### PR TITLE
Rat generator ordering

### PR DESCRIPTION
**Pull request by NickAragua**
The code changes are that a) saving the RAT generator state now consistently orders the models as well as the chassis (before it would randomly shuffle them around within the chassis) and b) the RAT generator now allows space stations to be generated.

The data changes are entries into the RAT generator for the various space stations. Here's a list as it's impossible to make head nor tail out of the RAT files themselves at the moment. Unless mentioned otherwise, the periphery always slightly "lags" behind IS in how common a station/variant is, and rarity has a brief ramp up period from when it was first released.

Alliance - civilian, rare
Alliance 3061 - civilian, more rare
Bastion - rare during star league era, ultra rare afterwards, regenerated post clan invasion
Bastion 3058 - rare post 3058
Bastion Clan - rare post 3090
Battlesat - common post 3056
Battlesat 3089 - common post 3089
Capitol - rare post 3050
Crucible - CC only, ultra rare, 3060+ (technically unique, but I took a liberal interpretation to give CC a little variety)
Drake SDS - rare WOB during Jihad period
Factory Small - civilian, common
Factory Medium/Large - civilian, rare
Factory Medium/Large 2750 - civilian, rarer, goes extinct during succession wars, slowly comes back 3039+
Habitat Small/Medium/Large/Large 2750 - common, 2750 version goes extinct during succession wars, slowly comes back 3039+
Hong Lung - CC only, pretty rare 3025+
M9 SDS - rare TH during star league era
Olympus - uncommon 2663+. 
Olympus 3072 - rare 3072+.
Olympus Clan - clan only, rare, 2950+
Pressurized Yard - small/medium/large huge - common, uncommon, rare, very rare
Unpressurized Yards - slightly more common than Pressurized Yards
Snowden Mining Station - TC only, Mk I rare 2390-2600, Standard rare 2600+, declines but never goes extinct

This will eventually be used to support an "Orbital Station" scenario type in StratCon, but can be used to generate somewhat flavorful batches of space stations

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/ForceGenerationOptionsPanel.java [hunk 1]
```diff
@@ -204,12 +204,8 @@ public ForceGenerationOptionsPanel(Use use) {
         c.weighty = 0.0;
         add(new JLabel(Messages.getString("RandomArmyDialog.UnitType")), c);
 
-        boolean restrictUnitType = use == Use.FORMATION_BUILDER;
-        
         for (int unitType : UNIT_TYPES) {
-            if ((unitType < UnitType.JUMPSHIP) || !restrictUnitType) {
-                cbUnitType.addItem(UnitType.getTypeName(unitType));
-            }
+            cbUnitType.addItem(UnitType.getTypeName(unitType));
```
Comment (neoancient) [inline megamek/src/megamek/client/ui/swing/ForceGenerationOptionsPanel.java:208]: Why did you change this?

Comment (NickAragua) [reply megamek/src/megamek/client/ui/swing/ForceGenerationOptionsPanel.java]: Reverted - I misread the logic.

File: megamek/src/megamek/client/ratgenerator/ChassisRecord.java [hunk 2]
```diff
@@ -45,6 +48,13 @@ public void addModel(ModelRecord model) {
         return models;
     }
     
+    public List<ModelRecord> getSortedModels() { 
+        List<ModelRecord> sortedModels = new ArrayList<>(models);
+        Collections.sort(sortedModels);
```
Comment (neoancient) [inline megamek/src/megamek/client/ratgenerator/ChassisRecord.java:53]: This sort does nothing. ModelRecord::compareTo sorts by the chassis name. All the models handled here have the same chassis and considered equal.

```suggestion
        sortedModels.sort(Comparator.comparing(ModelRecord::getModel));
```

File: megamek/src/megamek/client/ratgenerator/AbstractUnitRecord.java [hunk 3]
```diff
@@ -155,5 +155,9 @@ public static int parseUnitType(String typeName) {
 				return -1;
 		}
 	}
+	
+	public int compareTo(AbstractUnitRecord o2) {
+	    return this.getChassis().compareTo(o2.getChassis());
+	}
```
Comment (neoancient) [inline megamek/src/megamek/client/ratgenerator/AbstractUnitRecord.java:161]: The two classes that extend this one should sort on different fields. Only ChassisRecord uses the chassis, so there is no good reason to implement it this way in the base class.

**Review by neoancient — CHANGES_REQUESTED** [review]

**Review by HammerGS — APPROVED** [review]
I'm good with the Data stuff.

**Review by neoancient — APPROVED** [review]

**Pull request merged**